### PR TITLE
CI

### PR DESCRIPTION
**Pull request by cdmistman**
Why
===

initialize more comprehensive CI before sharing this repo with the communities

What changed
============

- added nix flake checks:
  - both of the binaries build
  - clippy
  - `cargo fmt`
  - `cargo nextest` (faster than `cargo test`)
  - check nix formatting (still using `alejandra`)
- use github actions instead of semaphore (it's more accessible for our open-source repos + DeterminateSystems easily integrates)
- formatted according to the updated checks

Test plan
=========

n/a

### CODE REVIEW

**Comment by cdmistman** [comment]
uh i'm gonna add an e2e test before re-opening

File: lib/genRegistry.nix
Comment (cdmistman) [inline lib/genRegistry.nix:1]: no logical changes, just formatting

File: src/bin/index/data.rs
Comment (cdmistman) [inline src/bin/index/data.rs:1]: no logical changes, just formatting

File: src/bin/index/main.rs
Comment (cdmistman) [inline src/bin/index/main.rs:1]: no logical changes, just formatting

**Comment by alexhulbert** [comment]
lgtm!

**Review by alexhulbert — APPROVED** [review]

**Pull request merged**